wasmtime-environ/src/cache.rs

```diff
@@ -505,3 +505,343 @@ impl<'de> Deserialize<'de> for JtOffsetsWrapper<'_> {
+                }
+            })
+            .collect();
+
+        let relocs = (0..rng.gen_range(1, 0x10))
+            .map(|i| {
+                vec![
+                    Relocation {
+                        reloc: binemit::Reloc::X86CallPCRel4,
+                        reloc_target: RelocationTarget::UserFunc(FuncIndex::new(i as usize * 42)),
+                        offset: i + rng.next_u32(),
+                        addend: 0,
+                    },
+                    Relocation {
+                        reloc: binemit::Reloc::Arm32Call,
+                        reloc_target: RelocationTarget::LibCall(ir::LibCall::CeilF64),
+                        offset: rng.gen_range(4, i + 55),
+                        addend: (42 * i) as i64,
+                    },
+                ]
+            })
+            .collect();
+
+        let trans = (4..rng.gen_range(4, 0x10))
+            .map(|i| FunctionAddressMap {
+                instructions: vec![InstructionAddressMap {
+                    srcloc: ir::SourceLoc::new(rng.gen()),
+                    code_offset: rng.gen(),
+                    code_len: i,
+                }],
+                start_srcloc: ir::SourceLoc::new(rng.gen()),
+                end_srcloc: ir::SourceLoc::new(rng.gen()),
+                body_offset: rng.gen(),
+                body_len: 0x31337,
+            })
+            .collect();
+
+        let value_ranges = (4..rng.gen_range(4, 0x10))
+            .map(|i| {
+                (i..i + rng.gen_range(4, 8))
+                    .map(|k| {
+                        (
+                            ir::ValueLabel::new(k),
+                            (0..rng.gen_range(0, 4))
+                                .map(|_| ValueLocRange {
+                                    loc: ir::ValueLoc::Reg(rng.gen()),
+                                    start: rng.gen(),
+                                    end: rng.gen(),
+                                })
+                                .collect(),
+                        )
+                    })
+                    .collect()
+            })
+            .collect();
+
+        let stack_slots = (0..rng.gen_range(0, 0x6))
+            .map(|_| {
+                let mut slots = ir::StackSlots::new();
+                slots.push(ir::StackSlotData {
+                    kind: ir::StackSlotKind::SpillSlot,
+                    size: rng.gen(),
+                    offset: rng.gen(),
+                });
+                slots.frame_size = rng.gen();
+                slots
+            })
+            .collect();
+
+        ModuleCacheData::from_tuple((
+            Compilation::new(funcs),
+            relocs,
+            trans,
+            value_ranges,
+            stack_slots,
+        ))
+    }
+
+    impl ModuleCacheEntry {
+        pub fn mod_cache_path(&self) -> &Option<PathBuf> {
+            &self.mod_cache_path
+        }
+    }
+
+    // cranelift's types (including SecondaryMap) doesn't implement PartialEq
```

Comment:
I know I originally suggested using this approach, but I actually think we can go back and implement `Eq`/`PartialEq` for `SecondaryMap`, with the approach of trimming the trailing default-value elements. That should hopefully allow the code here to be much simpler.

Reply:
As we discussed, I'll do it as a follow up refactor later, also implementing the serialization for `SecondaryMap`.